### PR TITLE
average LCO for carbon capture and cdr technologies

### PR DESCRIPTION
**Pull request by tabeado**
**Cost per tCO2:**
- for **all carbon capturing technologies that enter the CCUS-chain**: all cost are distributed across the captured co2 (thus output=cco2), only CTS cost are distributed across the share thereof that is injected (thus output=ico2). The total cost are calculated accordingly for both captured co2 (sum of capture cost) and for injected co2 (sum of captured + storage cost). In the csv output file, there is no further distinction between fossil, biogenic, and atmospheric options, as this would blow up the calculation.
- for **CDR technologies that do not enter the CCUS-chain**: introduction of the entity "sco2 = stored CO2". Currently only Enhanced Weathering, but already prepared to also accomodate biochar (&hopefully OAE) without fundamental changes.
- backwards compatibility: Script works also for runs prior to refactoring of the CDR module (Nov 2023). However, the fuel demand for dac & enhanced weathering would require significant addition of code that likely would not be worth it. These cost are thus (falsely) returned to be 0. 
- Note that *all* of the technologies' cost are accounted to CO2. Thus it is *not* the additional cost incurred through the carbon capture for the pe2se techs. 
 
**Smaller changes:**
- Make **discount rate** for average LCO explicit & align with marginal LCO calculation (from 0.06 to 0.05)
- Correct the **distribution of ccsinje cost** for teCCS and add **ccs-cost incl. adjustment cost**
- Add **secondary fuel cost** for all technologies
- Add **total cost** without & with all adjustment costs (also for pe2se technologies)
- Use same name for injected co2 (ico2) & sector (carbon management) for average and marginal LCO

**Levelized Cost of Enhanced Weathering**
Enhanced Weathering differs from other technologies as the period when cost are incurred for the performance of the technology is not the period when (all) of the removal is realized. I.e. the cost that are calculated according to the existing protocol are the cost of spreading the rock in t, which initiates removal in t, t+1 etc. at a declining rate. We can thus not simply divide by the removals from EW in t (reported by vm_emiCdrTeDetail) analogous to the other technologies.

There are then two ways of assigning cost of EW to removal:
(1) **the _cost of removal initiated_ in the period**: the cost of spreading are divided by the aggregated removal "initiated" thereby (i.e. the amount of rock spread * s33_co2_rem_pot). Note:
- fixed transport cost differ because different grades of transport distance need to be used.
- investment cost (which are negligible (max a few cents/tCO2) and thus mainly pro forma) differ because REMIND does not use the full standing capacity each time due to the negligible cost (may even not deploy in times where there is existing capacity)

(2) **the _cost of the removal realized_ in the period** through weathering of the rocks on the field: The thinking & calculation process is analogous to the general investment cost annuitization & distribution. This is currently not implemented. 

For documentation purpose, here an attempt to depict the overall logic: 
https://drive.google.com/file/d/1rMF-2TYvcULhBswGiRjHRymhO2tNFhgX/view?usp=sharing

### CODE REVIEW

**Comment by tabeado** [comment]
test results are here: /p/tmp/tabeado/remind2_clean/reportLCOE_enhancedWeathering

**Review by fschreyer — CHANGES_REQUESTED** [review]
I mainly looked at the small change that also affected the energy system technologies. I only had a superficial look at the carbon management part and did not spot anything in particular.

**Comment by tabeado** [comment]
@amerfort and @katarkow , could you have a look at the CDR-part please?

File: R/reportLCOE.R [hunk 1]
```diff
@@ -447,39 +548,129 @@ reportLCOE <- function(gdx, output.type = "both"){
...
                        paste0("LCOE|average|",pe2se$all_enty1,"|",pe2se$all_te, "|supply-side","|CO2 Cost")),
+          ### add curtailment cost
               setNames(te_curt_cost[,,pe2se$all_te],
                        paste0("LCOE|average|",pe2se$all_enty1,"|",pe2se$all_te, "|supply-side","|Curtailment Cost")),
-              #### Carbon Transport and storage ("ccsinje")
+          ### Total Cost
+              setNames((te_annual_inv_cost[,ttot_from2005,pe2se$all_te]+ te_annual_fuel_cost[,,pe2se$all_te] + te_annual_OMF_cost[,,pe2se$all_te] +
+                             te_annual_OMV_cost[,,pe2se$all_te] + te_annual_ccsInj_cost[,,pe2se$all_te] + te_annual_co2_cost[,,pe2se$all_te])/total_te_energy[,,pe2se$all_te] +
+                       (te_annual_stor_cost[,,pe2se$all_te] + te_annual_grid_cost[,,pe2se$all_te]) / total_te_energy_usable[,,pe2se$all_te] + te_curt_cost[,,pe2se$all_te],
+                       paste0("LCOE|average|",pe2se$all_enty1,"|",pe2se$all_te, "|supply-side","|Total Cost")),
```
[the remark this answers is not in the document]
Comment (tabeado) [reply R/reportLCOE.R:581]: Thank you for spotting this! I included the cost now

**Review by amerfort — APPROVED** [review]
Thanks for taking care of this! Numbers seem very reasonable.
In a next step, we should probably extract the CO2 capture part from the energy output. A first idea would be to subtract the price of the output fuel from the total LCO-CC. Let's discuss :)

**Review by fschreyer — APPROVED** [review]

**Pull request merged**